packages/components/src/components/post-collapsible/post-collapsible.tsx

```diff
@@ -0,0 +1,115 @@
+import { Component, Element, h, Method, Prop, State } from '@stencil/core';
+import { checkValidHTMLTag, getElementHeight, onTransitionEnd } from '../../utils';
+
+let nextId = 0;
+
+@Component({
+  tag: 'post-collapsible',
+  styleUrl: 'post-collapsible.scss',
+  shadow: true,
+})
+export class PostCollapsible {
+  /**
+   * If `true`, the element is initially collapsed otherwise it is displayed.
+   */
+  @Prop({ mutable: true }) collapsed?: boolean = false;
+
+  /**
+   * Defines the hierarchical level of the collapsible header within the headings structure.
+   */
+  @Prop() headingLevel?: number = 2;
+
+  @State() collapseClasses: string;
+  @State() collapseHeight: string | null = null;
+
+  @Element() host: HTMLElement;
+
+  hasHeader: boolean;
+  headingTag: string | undefined;
+  collapsibleId: string;
+  collapsibleElement: HTMLElement;
+
+  componentWillLoad() {
+    this.hasHeader = !!this.host.querySelector('[slot="header"]');
+    if (!this.hasHeader) {
+      console.warn('Be sure to bind the post-collapsible to its control using aria-controls and aria-expanded attributes. More information here: https://getbootstrap.com/docs/5.2/components/collapse/#accessibility');
+    }
+
+    this.headingTag = checkValidHTMLTag(`h${this.headingLevel}`, 'The post-collapsible element requires a heading level between 1 and 6.');
+
+    this.collapsibleId = this.host.id || `post-collapsible-${nextId++}`;
```

Comment:
I don't know why, but this produces `undefined--header` and `undefined-collapse` classes in the dom:
![image](https://user-images.githubusercontent.com/9716662/201678008-ca691d0f-af99-4276-a115-874aa61ce783.png)

Reply:
I could not reproduce this issue. Did you set anything special as an prop?

![image](https://user-images.githubusercontent.com/33580481/203514175-30d4073c-1bbb-4717-bf76-cabd3caf7d7c.png)

Reply:
I'm happy that I could not reproduce it anymore either. So you can ignore it...